src/machine/machined-varlink.c

```diff
@@ -596,6 +599,9 @@ static int list_image_one_and_maybe_read_metadata(sd_varlink *link, Image *image
         assert(link);
         assert(image);
 
+        if (!sd_varlink_is_processing_method(link))
+                return 0;
```

Comment:
hmm, I think we should probably not continue the looping at all once we fire sd_varlink_error().

I figure we should have sd_varlink_error_propagate() or so, which propagates an errno as return value.

so that `return sd_varlink_error_propagate(r, "io.systemd.Machine.NotAvailable", NULL);` would be a thing, where we first issue the error reply, but then also return from this function with the error code specified in the first arg.

Reply:
I decided to fix sd_varlink_error() itself instead.

See: #34946

With that I think this specific commit here can be dropped, no?

Reply:
> hmm, I think we should probably not continue the looping at all once we fire sd_varlink_error().

I support this. I asked the same question here https://github.com/systemd/systemd/pull/34882#discussion_r1822245664.

At the same time, changes to return error code from sd_varlink_error() is an even better alternative, which will prevent us from tracking places where we need to check `r < 0` and where `r != 0`